devimint/src/util.rs

```diff
@@ -21,19 +22,16 @@ fn kill(child: &Child) {
 /// NOTE: drop order is significant make sure fields in struct are declared in
 /// correct order it is generallly clients, process handle, deps
 #[derive(Debug, Clone)]
-pub struct ProcessHandle(Arc<ProcessHandleInner>);
+pub struct ProcessHandle(Arc<Mutex<ProcessHandleInner>>);
 
 impl ProcessHandle {
-    pub async fn kill(self) -> Result<()> {
-        let arc_process_handle_inner = self.0;
-        let mut process_handle_inner = match Arc::try_unwrap(arc_process_handle_inner) {
-            Ok(process_handler_inner) => process_handler_inner,
-            Err(_) => return Err(anyhow!("Cannot kill process because of clones")),
-        };
-        let mut child = std::mem::take(&mut process_handle_inner.child).unwrap();
-        info!(LOG_DEVIMINT, "killing {}", process_handle_inner.name);
-        kill(&child);
-        child.wait().await?;
+    pub async fn kill(&self) -> Result<()> {
+        let mut inner = self.0.lock().await;
+        if let Some(mut child) = inner.child.take() {
+            info!(LOG_DEVIMINT, "killing {}", inner.name);
+            kill(&child);
+            child.wait().await?;
```

Comment:
We should ignore the error here. If we're killing it, chances are high it will fail, and we don't really care anymore.

Reply:
We probably shouldn't even wait here in the first place. We've sent it a kill signal, we're done.

Reply:
No, we must be sure they are dead :wink:

Reply:
But the wait here doesn't not check if killed them right. It checks the return status of the command itself, which will be an error ... because we killed it. Unless it was already dead. No?

Reply:
It may or may be not an error. We are sending a SIGTERM, so the process can gracefully shutdown.

Reply:
If the process just ignores SIGTERM, then this won't return

Reply:
we could implement a timeout then retry will SIGKILL, but our processes are well behaved, so I don't think we need this for now

Reply:
Oh. You're right. I confused myself and thought we're sending SIGKILL already.

But I still think that most programs will terminate with an non-zero code on any signal:

![image](https://github.com/fedimint/fedimint/assets/9209/f322b4e4-a8b9-42f3-84b5-7f69c2fbff9a)

(red `>` means non-zero exit value, Ctrl+C sends a SIGINT).

So we should wait, but is there a point in making in an `Err`?